src/main/java/com/teragrep/cfe_39/consumers/kafka/BatchDistributionImpl.java

```diff
@@ -0,0 +1,180 @@
+ The mock consumer is activated for testing using the configuration file: readerKafkaProperties.getProperty("useMockKafkaConsumer", "false")*/
+
+public final class BatchDistributionImpl implements BatchDistribution {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(BatchDistributionImpl.class);
+
+    private final String topic;
+    private final DurationStatistics durationStatistics;
+    private final TopicCounter topicCounter;
+    private long lastTimeCalled;
+    private final ConfigurationImpl config;
+    private final Map<String, PartitionFileImpl> partitionFileMap;
+
+    public BatchDistributionImpl(
+            ConfigurationImpl config,
+            String topic,
+            DurationStatistics durationStatistics,
+            TopicCounter topicCounter
+    ) {
+        this.config = config;
+        this.topic = topic;
+        this.durationStatistics = durationStatistics;
+        this.topicCounter = topicCounter;
+        this.partitionFileMap = new HashMap<>();
+        this.lastTimeCalled = Instant.now().toEpochMilli();
+    }
+
+    /* Input parameter is a batch of RecordOffsetObjects from kafka. Each object contains a record and its metadata (topic, partition and offset).
+    * Distributes the received kafka record batch to PartitionFileImpl objects based on topic partition which the record originates from.
+    * */
+    @Override
+    public void accept(List<KafkaRecordImpl> batch) {
+        long thisTime = Instant.now().toEpochMilli();
+        long ftook = thisTime - lastTimeCalled;
+        topicCounter.setKafkaLatency(ftook);
+        if (LOGGER.isDebugEnabled()) {
+            LOGGER
+                    .debug(
+                            "Searching batch for <[{}]> with <{}> records took <{}> milliseconds. <{}> EPS. ", topic,
+                            batch.size(), (ftook), (batch.size() * 1000L / ftook)
+                    );
+        }
+        long batchBytes = 0L;
+        Timer timer = new Timer();
+        Timer.Context context = timer.time();
+
+        // Distribute the records of the batch to a PartitionFileImpl object based on partition from which the record originates from.
+        ListIterator<KafkaRecordImpl> recordOffsetListIterator = batch.listIterator();
+        while (recordOffsetListIterator.hasNext()) {
+            KafkaRecordImpl next = recordOffsetListIterator.next();
+            JsonObject recordOffset = JsonParser.parseString(next.offsetToJSON()).getAsJsonObject();
+            // If the PartitionFileImpl corresponding to the record's partition doesn't exist, create one.
+            if (!partitionFileMap.containsKey(recordOffset.get("partition").getAsString())) {
+                try {
+                    partitionFileMap
+                            .put(recordOffset.get("partition").getAsString(), new PartitionFileImpl(config, recordOffset));
```

Comment:
use static partitionFileFactory or so, so the object created is no longer configurable but gets always immutable configuration and only the recordOffset provided to the partitionFileFactory.partitionFor(recordOffset)